pkg/ccl/backupccl/restore_job.go

```diff
@@ -610,6 +610,11 @@ func restore(
 	// `restoreCtx` is used for orchestration logging. All operations that carry
 	// out work get their individual contexts.
 
+	// If there weren't any tables requested, then return early.
+	if len(tables) == 0 {
```

Comment:
This seems to cause the problems @pbardea. I added this because I was getting a divide by zero panic in later when i didnt have this, but it seems like returning here causes some system tables to not get restored. the divide by zero is raised by the splitAndScatter function. However, we can't just return in there, because other goroutines in this restore function depend on splitAndScatter to send some data over and gets stuck. So I'm not sure what to be returning early from. This became a problem after I moved the type jobs over to `publishTables()`

Reply:
I see. For tenant restore, we can have `len(tables)` be 0, but we still may be importing these spans. Does changing this to `len(spans) == 0` fix the issue?

Reply:
That seems to work